cfme/tests/infrastructure/test_host.py

```diff
@@ -312,3 +312,41 @@ def _finalize():
             actual_state, power_state
         ),
     )
+
+
+@test_requirements.general_ui
+def test_infrastructure_hosts_refresh_multi(
+    appliance, setup_provider, provider
+):
+    """
+    Polarion:
+        assignee: prichard
+        casecomponent: Infra
+        caseimportance: low
+        initialEstimate: 1/3h
+        testSteps:
+            1. Navigate to the Compute > Infrastructure > Hosts view.
+            2. Select at least 2 hosts (more if possible) by checking the box in upper left of
+               quadicons.
+            3. Click "Refresh Relationships and Power States" under the Configuration
+               dropdowm, and then click "OK" when prompted.
+        expectedResults:
+            1. Hosts view is displayed.
+            2.
+            3. "Refresh initiated for X Hosts from the CFME Database" is displayed in green
+               banner where "X" is the number of selected hosts. Properties for each host are
+               refreshed.
+    """
+    num_hosts = len(provider.data.get('hosts', {}))
+    if num_hosts < 2:
+        pytest.skip('not enough hosts to run test')
+    hosts_view = navigate_to(provider.collections.hosts, "All")
+    assert hosts_view.is_displayed
+    for h in hosts_view.entities.get_all():
+        h.check()
+    hosts_view.toolbar.configuration.item_select('Refresh Relationships and Power States',
+                                                 handle_alert=True)
+    hosts_view.flash.assert_success_message(
+        f'Refresh initiated for {num_hosts} Hosts from the CFME Database'
```

Comment:
I'm a bit concerned that refresh is made based on the number of hosts in the UI but `num_hosts` is determined based on the yaml file. Can you please explain?

Reply:
@john-dupuy well, here my concern is that for some reason not all hosts are in yamls (e.g. somebody added some test host), that way this test will fail.